game-app/game-core/src/main/java/games/strategy/triplea/settings/lobby/LobbySelectionView.java

```diff
@@ -0,0 +1,145 @@
+
+import com.google.common.annotations.VisibleForTesting;
+import games.strategy.triplea.settings.ClientSetting;
+import games.strategy.triplea.settings.SelectionComponent;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.swing.AbstractButton;
+import javax.swing.ButtonGroup;
+import javax.swing.JComponent;
+import javax.swing.JRadioButton;
+import javax.swing.JTextField;
+import org.triplea.java.StringUtils;
+import org.triplea.swing.JTextFieldBuilder;
+import org.triplea.swing.SwingComponents;
+import org.triplea.swing.jpanel.JPanelBuilder;
+import org.triplea.ui.events.queue.ControllerListener;
+import org.triplea.ui.events.queue.ViewClass;
+
+/**
+ * UI with: <br>
+ * 1) Radio buttons to select lobby, eg: prod / prerelease / local / other <br>
+ * 2) A text field for custom input of lobby URI, enabled only if 'other' is selected.<br>
+ *
+ * <p>The lobby text field is pre-filled and disabled for each radio button choice except for
+ * 'other' which will enable the uri field.
+ */
+class LobbySelectionView implements SelectionComponent<JComponent>, ViewClass {
+
+  private final LobbySelectionEventQueue lobbySelectionEventQueue;
+  private final JComponent uiComponent;
+  private final JTextField uriField;
+  private final List<JRadioButton> radioButtons;
+
+  LobbySelectionView(final LobbySelectionEventQueue lobbySelectionEventQueue) {
+    this.lobbySelectionEventQueue = lobbySelectionEventQueue;
+    final LobbySelectionViewData uiData = new LobbySelectionViewData();
+
+    uriField = new JTextFieldBuilder().build();
+    uriField.setText(uiData.getUriFieldValue());
+    uriField.setEnabled(uiData.getUriFieldEnabled());
+
+    final ButtonGroup buttonGroup = new ButtonGroup();
+    radioButtons =
+        // create a radio button for each of the lobby selection options
+        Stream.of(LobbySelectionViewData.LobbyChoice.values())
+            .map(
+                choice -> {
+                  final JRadioButton radioButton = new JRadioButton(choice.getDisplayString());
+                  buttonGroup.add(radioButton);
+                  radioButton.addActionListener(
+                      e ->
+                          lobbySelectionEventQueue.publishUiEvent(
+                              LobbySelectionEvents.ViewEvent.RADIO_BUTTON_CHANGED,
+                              readDataState()));
+
+                  return radioButton;
+                })
+            .collect(Collectors.toList());
+
+    uiComponent =
+        new JPanelBuilder()
+            .boxLayoutVertical()
+            .add(new JPanelBuilder().addAll(new ArrayList<>(radioButtons)).build())
+            .add(uriField)
+            .build();
+    updateUiToState(new LobbySelectionViewData());
+  }
+
+  @SuppressWarnings("SwitchStatementWithTooFewBranches")
+  @ControllerListener
+  public void updateUiToState(
```

Comment:
Note the usage of @ControllerListener to automatically have the event queue invoke this method.